dnf-behave-tests/createrepo_c/steps/lib/repodata.py

```diff
@@ -55,3 +55,32 @@ def verify_repomd_item_with_file(item, filepath, decompressed_filepath):
     if (a < b - 1 or a > b + 1):
         raise AssertionError("Timestamp of File: " + filepath + " doesn't match"
                              "timestamp in the repomd.xml:\n" + str(a) + " vs. " + str(b))
+
+def keys_do_not_differ(prim, flist, oth):
+    if set(prim.keys()) != set(flist.keys()):
+        raise AssertionError("Primary and Filelists have different package sets.")
+    if set(prim.keys()) != set(oth.keys()):
+        raise AssertionError("Primary and Other have different package sets.")
+
+
+def repodata_do_not_differ(prim1, prim2, flist1, flist2, oth1, oth2):
+        # Compare packages by checksums
+        if prim1.keys() != prim2.keys():
+            raise AssertionError("Primary repodata have different package sets.")
+
+        # Compare packages by name
+        if prim1.packages() != prim2.packages():
+            raise AssertionError("Primary repodata have different sets of package names.")
+
+        diff = prim1.diff(prim2)
+        if diff:
+            raise AssertionError("Primary repodata are different.\n"
+                                 "Difference: %s" % (diff))
+        diff = flist1.diff(flist2)
+        if diff:
+            raise AssertionError("Filelists repodata are different.\n"
+                                 "Difference: %s" % (diff))
+        diff = oth1.diff(oth2)
+        if diff:
+            raise AssertionError("Other repodata are different.\n"
+                                 "Difference: %s" % (diff))
```

Comment:
Oh, one more thing, as I mentioned before, I see no benefit in putting functions in a different ("lib") python module if you only use them in one place anyway. I'd prefer to put these at the top of the module you use them in, to save the reader having to go find them in a different place.